common/db_test.go

```diff
@@ -27,53 +27,42 @@ func TestDBLastSeenBlock(t *testing.T) {
 	defer dbh.Close()
 	defer dbraw.Close()
 
-	// sanity check default value
-	var val int64
-	var created_at string
-	var updated_at string
-	stmt := "SELECT value, updatedAt FROM kv WHERE key = 'lastBlock'"
-	row := dbraw.QueryRow(stmt)
-	err = row.Scan(&val, &created_at)
-	if err != nil || val != int64(0) {
-		t.Errorf("Unexpected result from sanity check; got %v - %v", err, val)
-		return
-	}
-	// set last updated at timestamp to sometime in the past
-	update_stmt := "UPDATE kv SET updatedAt = datetime('now', '-2 months') WHERE key = 'lastBlock'"
-	_, err = dbraw.Exec(update_stmt) // really should sanity check this result
-	if err != nil {
-		t.Error("Could not update db ", err)
-	}
-
-	// now test set
-	blkval := int64(54321)
-	err = dbh.SetLastSeenBlock(big.NewInt(blkval))
-	if err != nil {
-		t.Error("Unable to set last seen block ", err)
-		return
-	}
-	row = dbraw.QueryRow(stmt)
-	err = row.Scan(&val, &updated_at)
-	if err != nil || val != blkval {
-		t.Errorf("Unexpected result from value check; got %v - %v", err, val)
-		return
-	}
-	// small possibility of a test failure if we executed over a 1s boundary
-	if updated_at != created_at {
-		t.Errorf("Unexpected result from update check; got %v:%v", updated_at, created_at)
-		return
-	}
+	assert := assert.New(t)
+	require := require.New(t)
 
-	// test getter function
+	// When there are no headers, return nil
 	blk, err := dbh.LastSeenBlock()
-	if err != nil {
-		t.Error(err)
-		return
-	}
-	if blk.Int64() != blkval {
-		t.Errorf("Unexpected result from getter; expected %v, got %v", blkval, blk.Int64())
-		return
-	}
+	assert.Nil(err)
+	assert.Nil(blk)
+
+	// When there is a single header, return its number
+	h0 := defaultMiniHeader()
+	h0.Number = big.NewInt(100)
+	err = dbh.InsertMiniHeader(h0)
+	require.Nil(err)
```

Comment:
`require` will call `t.FailNow()` if the check evaluates to false (see the [require](https://godoc.org/github.com/stretchr/testify/require) package). Using `require` is similar to using `t.Fatal()` - if the check fails the test exits immediately rather than continuing to run which would be the case when using `assert` or `t.Error()`.

`require` is generally used in scenarios where a failing check would cause the rest of the test to provide little to no additional information (i.e. if the rest of the checks would always fail if the precondition specified by the `require` does not hold). In this case, we have already written unit tests for `InsertMiniHeader()` so we are assuming that it works correctly. So, the precondition that should hold is that `InsertMiniHeader()` does not return an error, otherwise the rest of the checks in the test will likely fail.

Reply:
Thanks for explaining ! 🥇